com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/quotalimit/QuotaLimitHandlerMediator.java

```diff
@@ -0,0 +1,126 @@
+		} catch (AxisFault ex) {
+			log.error(ex);
+		}
+    	
+		if (isQuotaEnabler) {
+		
+			if (operator.equalsIgnoreCase("nb")) {
+				operatorName =null;
+			}else if (operator.equalsIgnoreCase("sb")) {
+				operatorName=(String)messageContext.getProperty("OPERATOR_ID");			
+			}
+			
+			serviceProvider=(String)messageContext.getProperty("USER_ID");
+			application=(String)messageContext.getProperty("APPLICATION_ID");
+			apiName=(String)messageContext.getProperty("API_NAME");
+			
+			APIService apiService = new APIService();
+	    	
+				try {
+		            QuotaLimits QuotaLimit=apiService.checkQuotaLimit(serviceProvider,application,apiName,operatorName);
+		            QuotaLimits currentQuotaLimit=apiService.currentQuotaLimit(serviceProvider,application,apiName,operatorName);
+
+		            if (QuotaLimit.getAppLimit()<=currentQuotaLimit.getAppLimit()) {
+		                setErrorInContext(messageContext,"POL1001","The %1 quota limit for this Application has been exceeded","QuotaLimit","400","POLICY_EXCEPTION");
+		                messageContext.setProperty("INTERNAL_ERROR", "true");
+					}
+		            if (QuotaLimit.getSpLimit()<=currentQuotaLimit.getSpLimit()) {
+		                setErrorInContext(messageContext,"POL1001","The %1 quota limit for this Service Provider has been exceeded","QuotaLimit","400","POLICY_EXCEPTION");
+		                messageContext.setProperty("INTERNAL_ERROR", "true");
+					}
+
+		            if (QuotaLimit.getApiLimit()<=currentQuotaLimit.getApiLimit()) {
+		                setErrorInContext(messageContext,"POL1001","The %1 quota limit for this API has been exceeded","QuotaLimit","400","POLICY_EXCEPTION");
+		                messageContext.setProperty("INTERNAL_ERROR", "true");
+					}
+
+		        } catch (Exception e) {
+		            log.error("Error occurred while calling QuotaLimitCheckMediator" ,e);
+		            setErrorInContext(messageContext,"SVC0001","A service error occurred. Error code is %1","An internal service error has occured. Please try again later.","500", "SERVICE_EXCEPTION");
+		            messageContext.setProperty("INTERNAL_ERROR", "true");
+		        }
+			}
+		
+        return true;
+    }
+    
+	@SuppressWarnings("rawtypes")
+	public static boolean isQuotaEnabler(MessageContext context) throws AxisFault {
+		boolean quotaEnabler = false;
+		try {
+			org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) context).getAxis2MessageContext();
+			Object headers = axis2MessageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+			if (headers != null && headers instanceof Map) {
+				Map headersMap = (Map) headers;
+				String jwtparam = (String) headersMap.get("x-jwt-assertion");
+				String[] jwttoken = jwtparam.split("\\.");
+				String jwtbody = Base64Coder.decodeString(jwttoken[1]);
+				JSONObject jwtobj = new JSONObject(jwtbody);
+				Iterator<String> keys = jwtobj.keys();
+				while( keys.hasNext() ) {
```

Comment:
directly pick the key which contains list of roles and iterate over that one only